### ISSUE TITLE
Viewer oscillate with sidebar hiding and showing

### ISSUE THREAD

**Issue by lingxiao14122** [stephanrauh/ngx-extended-pdf-viewer#2689]
**Check this first**
- Does the bug also happen on https://mozilla.github.io/pdf.js/web/viewer.html?
No
- Does the bug also happen on https://pdfviewer.net/simple?
Yes

**Describe the bug**
1. Go to [https://pdfviewer.net/extended-pdf-viewer/simple](https://pdfviewer.net/extended-pdf-viewer/simple)
2. Open a single page pdf
3. Select zoom "Automatic Zoom" or "Page Width"
4. Reduce window width untill sidebar about to hide
5. At a certain point the oscillate will start

**Version info**
- Version of ngx-extended-pdf-viewer 19.0.3

**Demo PDF file**
[dummy.pdf](https://github.com/user-attachments/files/18140163/dummy.pdf)

**Screen Capture**

https://github.com/user-attachments/assets/acc4bc32-2bc1-4e12-a28a-ec207062d54f

**Comment by stephanrauh** [comment]
I've watch your video, so the bug is obviously there, but I can't reproduce it. I've tried to do so with Google Chrome on the same URL and with the same file as you did. What am I missing?

**Comment by lingxiao14122** [comment]
Could you try again in this sandbox? https://codesandbox.io/p/devbox/8s9ppq
or you might want zoom out on browser, the screen capture was on a bigger screen

**Comment by stephanrauh** [comment]
@anlampe Can you find out which div has the inner scrollbar, please? Probably it's either #mainContainer, #viewerContainer or #viewer. Then you can hide the scrollbar in your global `styles.(s)css`:

```css
ngx-extended-pdf-viewer #viewerContainer {
   overflow-y: hidden !important;
}
```
If that solves the issue, and if you don't see any disadvantages of hiding the scrollbar, I can implement a cleaner version of this workaround in ngx-extended-pdf-viewer.

**Comment by anlampe** [comment]
> That's probably the reason why I can't reproduce the issue.

Yes, these dynamically appearing scrollbars on Windows seem to cause the issue.

Changing the vertical overflow of `#viewcontainer` from auto to scroll (hidden is not an option as it prevents users from scrolling longer documents) so that the scrollbar is "static" solves the flickering. One other option would be working with `scrollbar-gutter: stable;` or `scrollbar-gutter: stable both-edges;`. This would also prevent these unwanted layout shifts when the scrollbar appears.

**Comment by stephanrauh** [comment]
I've been able to reproduce the issue on Windows. To be honest, the Mac approach is a bit weird. It may look sleek and elegant, but in the case of this viewer, it means you can't use the scrollbar because you don't know it's there.

I was even able to sort of reproduce the issue on the pdf.js showcase: https://mozilla.github.io/pdf.js/web/viewer.html. It doesn't start flickering, but when you resize the window slowly, there's a short time when the scrollbar vanishes and re-appears when you make the window smaller. @Timvandermeij Maybe that's interesting to you, too. You don't have to do anything, but you should be aware there's a corner-case when pdf.js behaves a bit odd.

As for the Mac: the hidden scrollbar seems to be an old issue: https://stackoverflow.com/questions/7492062/css-overflow-scroll-always-show-vertical-scroll-bar

**Comment by stephanrauh** [comment]
I've decided to use `scrollbar-gutter: stable;`. It seems to be the best compromise, and I didn't see any disadvantage, except for a slightly smaller PDF file.

**Comment by stephanrauh** [comment]
Your bugfix has landed with version 22.2.4.

Enjoy!
Stephan
